fe/fe-core/src/main/java/com/starrocks/catalog/MaterializedView.java

```diff
@@ -0,0 +1,494 @@
+            return type;
+        }
+
+        public void setType(RefreshType type) {
+            this.type = type;
+        }
+
+        public AsyncRefreshContext getAsyncRefreshContext() {
+            return asyncRefreshContext;
+        }
+
+        public void setAsyncRefreshContext(AsyncRefreshContext asyncRefreshContext) {
+            this.asyncRefreshContext = asyncRefreshContext;
+        }
+
+        public long getLastRefreshTime() {
+            return lastRefreshTime;
+        }
+
+        public void setLastRefreshTime(long lastRefreshTime) {
+            this.lastRefreshTime = lastRefreshTime;
+        }
+    }
+
+    public enum MaterializedViewState {
+        NORMAL,
+        ROLLUP,
+        SCHEMA_CHANGE,
+        RESTORE,
+        RESTORE_WITH_LOAD,
+        /*
+         * this state means table is under PENDING alter operation(SCHEMA_CHANGE or ROLLUP), and is not
+         * stable. The tablet scheduler will continue fixing the tablets of this table. And the state will
+         * change back to SCHEMA_CHANGE or ROLLUP after table is stable, and continue doing alter operation.
+         * This state is a in-memory state and no need to persist.
+         */
+        WAITING_STABLE
+    }
+    @SerializedName(value = "dbId")
+    protected long dbId;
+
+    @SerializedName(value = "clusterId")
+    protected int clusterId;
+
+    @SerializedName(value = "state")
+    protected MaterializedViewState state;
+
+    // index id -> index meta
+    @SerializedName(value = "indexIdToMeta")
+    protected Map<Long, MaterializedIndexMeta> indexIdToMeta = Maps.newHashMap();
+    // index name -> index id
+    @SerializedName(value = "indexNameToId")
+    protected Map<String, Long> indexNameToId = Maps.newHashMap();
+
+    @SerializedName(value = "keysType")
+    protected KeysType keysType;
+    @SerializedName(value = "partitionInfo")
+    protected PartitionInfo partitionInfo;
+    @SerializedName(value = "idToPartition")
+    protected Map<Long, Partition> idToPartition = new HashMap<>();
+    @SerializedName(value = "nameToPartition")
+    protected Map<String, Partition> nameToPartition = Maps.newTreeMap(String.CASE_INSENSITIVE_ORDER);
+
+    @SerializedName(value = "defaultDistributionInfo")
+    protected DistributionInfo defaultDistributionInfo;
+
+    @SerializedName(value = "baseIndexId")
+    protected long baseIndexId = -1;
+
+    @SerializedName(value = "tableProperty")
+    protected TableProperty tableProperty;
+
+    @SerializedName(value = "refreshScheme")
+    private MvRefreshScheme refreshScheme;
+
+    @SerializedName(value = "baseTables")
+    private List<Long> baseTableIds;
+
+    @SerializedName(value = "active")
+    private boolean active;
+
+    // TODO: partition expression
+
+    // TODO: now it is original definition sql
+    // for show create mv, constructing refresh job(insert into select)
+    @SerializedName(value = "viewDefineSql")
+    private String viewDefineSql;
```

Comment:
This defineSql need to be expose for generate insert overwrite sql

Reply:
ok. I will add one api